src/main/java/com/example/spring/controller/BookController.java

```diff
@@ -34,7 +35,7 @@ public BookDto getBookById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
```

Comment:
Add `@Valid` to updateBook endpoint

Also, you can validate Long id by using `@Positive`. In such case add `@Validated` under the controller